WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -12,54 +12,35 @@ namespace WalletWasabi.Blockchain.TransactionOutputs;
 public class CoinsRegistry : ICoinsView
 {
 	/// <remarks>Guarded by <see cref="Lock"/>.</remarks>
-	private HashSet<SmartCoin> Coins { get; } = new();
+	private Dictionary<uint256, Dictionary<OutPoint, SmartCoin>> CoinsByTransactionsIds { get; } = new();
```

Comment:
> To clarify this: So each coin (unspent) is indexed by its txid and its transaction output number. Is that right?

It is true, but not only unspent, I am storing all coins in this data structure, whether they are spent or unspent.

> If it is true, then maybe Dictionary<uint256, Dictionary<uint, SmartCoin>> CoinsByTransactionsIds { get; } = new(); feels more natural because that OutPoint contains a txid again

Yes, in fact I am already relying on this here:
https://github.com/zkSNACKs/WalletWasabi/pull/11516/files#diff-e8d1b0f75371732cf879a147147d6936d13415fc9c758dd0b727a847915afbeaR185

Maybe it's a good idea, I need to check how it affects performances